crates/build/src/lib.rs

```diff
@@ -611,8 +635,7 @@ pub fn execute(args: ExecuteArgs) -> Result<BuildResult> {
     let build =
         || -> Result<(Option<OptimizationResult>, BuildInfo, PathBuf, BuildSteps)> {
             let mut build_steps = BuildSteps::new();
-            let pre_fingerprint =
-                Fingerprint::try_from_path(&crate_metadata.original_wasm)?;
+            let (pre_fingerprint, target_file_path) = Fingerprint::new(&crate_metadata)?;
```

Comment:
Let's avoid this tuple return by constructing `target_file_path` here and passing it to `Fingerprint::new`

Reply:
I moved the path up into `CrateMetadata`.